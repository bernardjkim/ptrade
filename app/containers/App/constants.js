/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

// CREATE SESSION
export const CREATE_SESSION = 'app/App/CREATE_SESSION';
export const CREATE_SESSION_SUCCESS = 'app/App/CREATE_SESSION_SUCCESS';
export const CREATE_SESSION_ERROR = 'app/App/CREATE_SESSION_ERROR';

// DELETE SESSION
// TODO: is it enough to just delete the client-side token? or do i need to
// black list token on the backend?
export const DELETE_SESSION = 'app/App/DELETE_SESSION';
export const DELETE_SESSION_SUCCESS = 'app/App/DELETE_SESSION_SUCCESS';
export const DELETE_SESSION_ERROR = 'app/App/DELETE_SESSION_ERROR';

export const LOAD_TOKEN = 'app/App/LOAD_TOKEN';
