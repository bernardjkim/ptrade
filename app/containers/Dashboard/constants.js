/*
 *
 * Dashboard constants
 *
 */

export const CHANGE_SEARCH = 'app/Dashboard/CHANGE_SEARCH';
export const CHANGE_TIME_FRAME = 'app/Dashboard/CHANGE_TIME_FRAME';

export const LOAD_STOCK_DATA = 'app/Dashboard/LOAD_STOCK_DATA';
export const LOAD_STOCK_DATA_SUCCESS = 'app/Dashboard/LOAD_STOCK_DATA_SUCCESS';
export const LOAD_STOCK_DATA_ERROR = 'app/Dashboard/LOAD_STOCK_DATA_ERROR';
