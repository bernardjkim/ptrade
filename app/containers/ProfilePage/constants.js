/*
 *
 * ProfilePage constants
 *
 */

export const CHANGE_TIME_FRAME = 'app/ProfilePage/CHANGE_TIME_FRAME';
export const SET_PORTFOLIO_VALUE = 'app/ProfilePage/SET_PORTFOLIO_VALUE';

export const LOAD_BALANCE = 'app/ProfilePage/LOAD_BALANCE';
export const LOAD_BALANCE_SUCCESS = 'app/ProfilePage/LOAD_BALANCE_SUCCCESS';
export const LOAD_BALANCE_ERROR = 'app/ProfilePage/LOAD_BALANCE_ERROR';

export const LOAD_POSITIONS = 'app/ProfilePage/LOAD_POSITIONS';
export const LOAD_POSITIONS_SUCCESS = 'app/ProfilePage/LOAD_POSITIONS_SUCCESS';
export const LOAD_POSITIONS_ERROR = 'app/ProfilePage/LOAD_POSITIONS_ERROR';

export const LOAD_CHART = 'app/ProfilePage/LOAD_CHART';
export const LOAD_CHART_SUCCESS = 'app/ProfilePage/LOAD_CHART_SUCCESS';
export const LOAD_CHART_ERROR = 'app/ProfilePage/LOAD_CHART_ERROR';
