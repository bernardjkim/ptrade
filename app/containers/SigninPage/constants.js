/*
 *
 * SigninPage constants
 *
 */

export const CHANGE_INPUT = 'app/SigninPage/CHANGE_INPUT';
