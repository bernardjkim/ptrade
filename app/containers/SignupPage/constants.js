/*
 *
 * SignupPage constants
 *
 */

export const CHANGE_INPUT = 'app/SignupPage/CHANGE_INPUT';

export const CREATE_USER = 'app/SignupPage/CREATE_USER';
export const CREATE_USER_SUCCESS = 'app/SignupPage/CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'app/SignupPage/CREATE_USER_ERROR';
