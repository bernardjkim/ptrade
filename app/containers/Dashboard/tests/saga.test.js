/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { all, put, takeLatest } from 'redux-saga/effects';
import { LOAD_CHART, LOAD_QUOTE } from '../constants';
import {
  chartLoaded,
  chartLoadingError,
  quoteLoaded,
  quoteLoadingError,
} from '../actions';
import dashboard, { getChart, getQuote } from '../saga';

const symbol = 'AAPL';

describe('getChart Saga', () => {
  let getDataGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getDataGenerator = getChart();

    const selectDescriptor = getDataGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getDataGenerator.next(symbol).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('requests the data successfully with 1d data', () => {
    const timeFrame = 0;
    const response = [
      { date: '09:30', value: 100 },
      { date: '09:35', value: 101 },
    ];
    const data = [
      { minute: '09:30', marketAverage: 100 },
      { minute: '09:35', marketAverage: 101 },
    ];

    getDataGenerator.next(timeFrame);
    const putDescriptor = getDataGenerator.next(data).value;
    expect(putDescriptor).toEqual(put(chartLoaded(response, symbol)));
  });

  it('requests the data successfully with 1m data', () => {
    const timeFrame = 1;
    const response = [
      { date: 'Oct 01 2018', value: 100 },
      { date: 'Oct 02 2018', value: 101 },
    ];
    const data = [
      { date: '2018-10-01', close: 100 },
      { date: '2018-10-02', close: 101 },
    ];

    getDataGenerator.next(timeFrame);
    const putDescriptor = getDataGenerator.next(data).value;
    expect(putDescriptor).toEqual(put(chartLoaded(response)));
  });

  it('should call the chartLoadingError action if the response errors', () => {
    const timeFrame = 0;
    const response = new Error('Some error');

    getDataGenerator.next(timeFrame);
    const putDescriptor = getDataGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(chartLoadingError(response)));
  });
});

describe('getQuote Saga', () => {
  let getDataGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getDataGenerator = getQuote();

    const selectDescriptor = getDataGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getDataGenerator.next(symbol).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('requests the quote successfully', () => {
    const response = { quote: {} };
    const data = { quote: {} };
    const putDescriptor = getDataGenerator.next(data).value;
    expect(putDescriptor).toEqual(put(quoteLoaded(response)));
  });

  it('should call the chartLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getDataGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(quoteLoadingError(response)));
  });
});

describe('Saga', () => {
  const dashboardSaga = dashboard();

  it('should start task to watch for LOAD_CHART/LOAD_QUOTE action', () => {
    const takeLatestDescriptor = dashboardSaga.next().value;
    expect(takeLatestDescriptor).toEqual(
      all([takeLatest(LOAD_CHART, getChart), takeLatest(LOAD_QUOTE, getQuote)]),
    );
  });
});
