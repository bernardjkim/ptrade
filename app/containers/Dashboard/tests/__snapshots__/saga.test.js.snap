// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getChart Saga requests the data successfully with 1d data 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getChart Saga requests the data successfully with 1d data 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getChart Saga requests the data successfully with 1m data 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getChart Saga requests the data successfully with 1m data 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getChart Saga should call the chartLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getChart Saga should call the chartLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getQuote Saga requests the quote successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getQuote Saga requests the quote successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.iextrading.com/1.0/stock/AAPL/quote",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getQuote Saga should call the chartLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getQuote Saga should call the chartLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.iextrading.com/1.0/stock/AAPL/quote",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
